onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -98,6 +87,31 @@ static bool IsNeighborNodeExpectedTypes(Node::NodeConstIterator start, const Nod
   return start == end;
 }
 
+static void RemoveWithSharedNodeInTop(Graph& graph, const std::vector<NodeIndex>& nodes_to_remove) {
+  std::unordered_set<NodeIndex> nodes_to_remove_set(nodes_to_remove.begin(), nodes_to_remove.end());
+
+  for (const NodeIndex index : nodes_to_remove) {
+    Node* node = graph.GetNode(index);
+    int child_node_to_remove_count = 0;
+    int child_node_to_stay_count = 0;
+    for (Node::NodeConstIterator it = node->OutputEdgesBegin(), end = node->OutputEdgesEnd(); it != end; ++it) {
+      std::unordered_set<NodeIndex>::const_iterator got = nodes_to_remove_set.find((*it).Index());
+      if (got == nodes_to_remove_set.end()) {
```

Comment:
If some nodes in nodes_to_remove_set are not removed in the end, the counters here are not accurate.

Reply:
The nodes_to_remove_set will not change because if shared node's children are removed it could be serve as an output node which is valid to remove in the end. 
Never mind I'll rework to use graph_utils::RemoveNodesWithOneOutputBottomUp